hazelcast/src/main/java/com/hazelcast/replicatedmap/impl/record/EntrySetIteratorFactory.java

```diff
@@ -49,6 +50,18 @@ public boolean hasNext() {
 
             @Override
             public Map.Entry<K, V> next() {
+                if (entry == null) {
+                    while (iterator.hasNext()) {
```

Comment:
- Isn't this loop skipping entries? In hasNext() method, it calls .next(), checks the entry and returns true. After that, it does not use that entry, instead of it, it makes a new .next() call. Therefore it skips an entry. Is my understanding wrong? 
- Isn't if(!testEntry(entry)) in next() method redundant? hasNext() method already validates an entry before returning.
- Shouldn't we set null to entry after next() method returns a value?